File: 22.10/169.go
func majorityElement(nums []int) int {
    m := make(map[int]int)
    
    for _, i := range(nums) {
        m[i]++
    }
    
    max := nums[0]
    for key, val := range(m) {
        if m[max] < val {
            max = key
        }
    }
    
    return max
}

File: typescript/23.08/169.ts
function majorityElement(nums: number[]): number {
    const visit = {}
    for (const x of nums) {
        if (!visit[x]) {
            visit[x] = 0
        }
        visit[x]++
        if (visit[x] > nums.length/2) {
            return x
        }
    }
};
